src/ast.zig

```diff
@@ -54,6 +55,32 @@ fn fullPtrTypeComponents(tree: Ast, info: full.PtrType.Components) full.PtrType
     return result;
 }
 
+fn findMatchingRBrace(token_tags: []const TokenTag, l_brace: Ast.TokenIndex) Ast.TokenIndex {
+    std.debug.assert(token_tags[l_brace] == TokenTag.l_brace);
+
+    const start = l_brace + 1;
+    var depth: i32 = 0;
+    var offset: Ast.TokenIndex = 0;
+
+    for (token_tags[start..], 0..) |tag, i| {
+        if (tag == TokenTag.l_brace) {
+            depth += 1;
+        }
+        if (tag == TokenTag.r_brace) {
+            if (depth == 0) {
+                offset = @intCast(i);
+                break;
+            }
+            depth -= 1;
+        }
+    }
+
+    std.debug.assert(depth == 0);
+    std.debug.assert(token_tags[start + offset] == std.zig.Token.Tag.r_brace);
```

Comment:
Yes, I'm glad you brought this up since I forgot to ask about it. `lastToken` does not return an error, so I was not sure how to fail gracefully in these failure situations. Is there some null-like token index that should be returned? Should `lastToken` be modified to return an error? Should I just return the last token I see and hope for the best?

Reply:
changing the return type of `lastToken` should probably be avoided because it is used in many places. If you were to return null or an error, the caller of `lastToken` would then need to figure out a suitable last token which isn't easy to do. Given that this only happens on code that is mostly syntactical nonsense, there is no harm in returning a "bad" result from `lastToken`.
Here is my sugesstion: 
change the return type of `findMatchingRBrace` to `?Ast.TokenIndex` and fall back to the previous implementation in `lastToken` if `findMatchingRBrace` returns null.
Returns the very last token of the source file would also be possible.

Reply:
I agree, there is not really much the caller could do in the case of broken syntax even if they got an error. I think I favor returning the last token vs falling back to the old implementation. Neither one is going to do the right thing, but the former is at least a little more clear about that.